client/app/src/main/java/com/example/criengine/Activities/RegisterAccount.java

```diff
@@ -100,32 +109,81 @@ protected void onCreate(Bundle savedInstanceState) {
                 firstName = firstnameField.getText().toString();
                 lastName = lastnameField.getText().toString();
-
-                // TODO: Create new account in database and log into it.
-
-                // Proceed into the app.
-                Intent intent = new Intent(v.getContext(), RootActivity.class);
-                v.getContext().startActivity(intent);
+                FirebaseFirestore.getInstance().collection("users").whereEqualTo("username",username).get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        if (task.isSuccessful()) {
+                            QuerySnapshot query = task.getResult();
+                            if (query != null && query.isEmpty()) {
+                                mAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                                    @Override
+                                    public void onComplete(@NonNull Task<AuthResult> task) {
+                                        if (task.isSuccessful() && task.getResult() != null) {
+                                            AuthResult authResult = task.getResult();
+                                            FirebaseUser user = authResult.getUser();
+                                            Profile profile = new Profile(user.getUid(), email, username, firstName, lastName);
+                                            DatabaseWrapper dbw = new DatabaseWrapper(user);
+                                            dbw.addProfile(profile).addOnCompleteListener(new OnCompleteListener<Void>() {
+                                                @Override
+                                                public void onComplete(@NonNull Task<Void> task) {
+                                                    Intent intent = new Intent(v.getContext(), RootActivity.class);
+                                                    v.getContext().startActivity(intent);
+                                                }
+                                            });
+                                        } else {
+                                            Log.d(TAG, "Get Failure: " + task.getException());
+                                            enableSubmit(true);
+                                            return;
+                                        }
+                                    }
+                                });
+                            } else {
+                                usernameField.setError("This username is already taken!");
+                                enableSubmit(true);
+                            }
+                        } else {
+                            Log.d(TAG, "Get Failure: " + task.getException());
```

Comment:
Also, do we want logs in the final draft?

Reply:
Any of these failures mean that a netowrk request has failed. In a perfect world they shouldn't, and if they appear we already failed, its good to know how.